components/TicTacToe.jsx

```diff
@@ -7,22 +7,64 @@ export const TicTacToe = () => {
     border: "1px solid",
     boxSizing: "content-box",
   };
+  const cols = 3;
+  const rows = 3;
   const pieces = ["X", "O"];
   const [turn, setTurn] = useState(0);
   const [grid, setGrid] = useState(Array(9).fill(null));
+  const [win, setWin] = useState("in progress");
+
+  const notClicked = (i) => {
+    if (pieces.indexOf(grid[i]) < 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const gridIndex = (x, y) => {
+    return cols * x + y;
+  };
+
+  const checkWinHorizontal = () => {
+    for (let x = 0; x < rows; x += 1) {
+      for (let y = 0; y < cols; y += 1) {
+        if (
+          grid[gridIndex(x, y)] == grid[gridIndex(x, y + 1)] &&
+          grid[gridIndex(x, y + 1)] == grid[gridIndex(x, y + 2)] &&
+          grid[gridIndex(x, y)] != null
+        ) {
+          return true;
```

Comment:
Recommend only single return statement

https://softwareengineering.stackexchange.com/questions/118703/where-did-the-notion-of-one-return-only-come-from

Reply:
Similar to the above function, since you're not running any code inside of the blocks, just directly return the expression in the `if` clause.